### ISSUE TITLE
Condition StringNotEqualsIfExists incorrectly failing?

### ISSUE THREAD

**Issue by gbudge** [aws-cloudformation/cfn-lint#3783]
### CloudFormation Lint Version

cfn-lint 1.18.0

### What operating system are you using?

Linux

### Describe the bug

According to the [Conditions_IfExists](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_IfExists) documentation, `StringNotEqualsIfExists` should be a valid Condition however cfn-lint produces the following finding:

`[cfn-lint] E3510: 'StringNotEqualsIfExists' does not match any of the regexes: 'ForAllValues:^(Not)?IpAddress$', 'ForAllValues:^Arn(Not)?Equals$', 'ForAllValues:^Arn(Not)?Like$', 'ForAllValues:^Date(Not)?Equals$', 'ForAllValues:^Number(Less|Greater)Than(Equals)?$', 'ForAllValues:^Number(Not)?Equals$', 'ForAllValues:^String(Not)?Equals(IgnoreCase)?$', 'ForAllValues:^String(Not)?Like$', 'ForAnyValues:^(Not)?IpAddress$', 'ForAnyValues:^Arn(Not)?Equals$', 'ForAnyValues:^Arn(Not)?Like$', 'ForAnyValues:^Date(Not)?Equals$', 'ForAnyValues:^Number(Less|Greater)Than(Equals)?$', 'ForAnyValues:^Number(Not)?Equals$', 'ForAnyValues:^String(Not)?Equals(IgnoreCase)?$', 'ForAnyValues:^String(Not)?Like?$', '^(Not)?IpAddress(Exists)?$', '^Arn(Not)?Equals(Exists)?$', '^Arn(Not)?Like(Exists)?$', '^Date(Not)?Equals(Exists)?$', '^Number(Less|Greater)Than(Equals)?(Exists)?$', '^Number(Not)?Equals(Exists)?$', '^String(Not)?Equals(IgnoreCase)?(Exists)?$', '^String(Not)?Like(Exists)?$'`

### Expected behavior

The `StringNotEqualsIfExists` Condition operator passes validation.

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: '2010-09-09'

Description: Demo policy

Parameters:
  NotebookRole:
    Type: String
    Description: The role name
    Default: DemoNotebookRole

Resources:
  NotebookPolicy:
    Type: AWS::IAM::Policy
    Properties:
      PolicyName: DemoNotebookPolicy
      Roles: 
        - !Ref NotebookRole
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - 'sagemaker:*'
            Resource: '*'
            Condition:
              StringNotEqualsIfExists:
                'aws:RequestTag/demo-tag': 'true'
```

**Comment by kddejong** [comment]
Fixed in [v1.18.1](https://github.com/aws-cloudformation/cfn-lint/releases/tag/v1.18.1)
